src/v/redpanda/admin_server.cc

```diff
@@ -121,6 +121,24 @@ admin_server::admin_server(
 ss::future<> admin_server::start() {
     configure_metrics_route();
     configure_admin_routes();
+    _server._routes.register_exeption_handler(
+      [](std::exception_ptr eptr) -> std::unique_ptr<ss::httpd::reply> {
+          try {
+              std::rethrow_exception(eptr);
+          } catch (ss::httpd::base_exception& ex) {
+              const auto status = static_cast<
+                std::underlying_type_t<ss::httpd::reply::status_type>>(
+                ex.status());
+              if (status >= 400 && status <= 499) {
+                  vlog(logger.warn, "exception encountered - {}", eptr);
+              } else if (status >= 500) {
+                  vlog(logger.error, "exception encountered - {}", eptr);
```

Comment:
From the looks of the failed tests, there's likely a category of "expected" retriable server errors we ought to be mindful of in catching. Looks like mostly `ss::httpd::reply::status_type::service_unavailable`, though maybe there's more.

Reply:
Right, 503 is not an error and shouldn't be logged as one.